From 139d7c2c9f8f9f21304e2872f532bf7560d7cf07 Mon Sep 17 00:00:00 2001
From: Stuart Menefy <stuart.menefy@codasip.com>
Date: Thu, 18 Jan 2024 11:43:11 +0000
Subject: [PATCH 03/15] riscv: dts: Add codasip devicetree

Signed-off-by: Stuart Menefy <stuart.menefy@codasip.com>
---
 arch/riscv/boot/dts/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/riscv/boot/dts/Makefile b/arch/riscv/boot/dts/Makefile
index 72030fd727af..b03fa5a57bb1 100644
--- a/arch/riscv/boot/dts/Makefile
+++ b/arch/riscv/boot/dts/Makefile
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 subdir-y += allwinner
 subdir-y += canaan
+subdir-y += codasip
 subdir-y += microchip
 subdir-y += renesas
 subdir-y += sifive
-- 
2.39.3

